src/rpdk/core/contract/resource_client.py

```diff
@@ -69,6 +69,30 @@ def override_properties(document, overrides):
     return document
 
 
+def prune_properties_except_identifier(model, paths):
```

Comment:
Why does this function modify the document in place?  You are already passing a copy in here, I don't see why this should be the responsibility of the caller.  

Also, this seems like a generally useful utility function, and it could be more generically named `prune_properties` with the parameter called `paths_to_retain` or something.

The reason I bring this up is because the read handler should be able to be invoked with any single `additional_identifier` as well.  I'm okay with adding contract tests for that in a separate PR btw

Reply:
I was actually following what similar prune methods  are doing. But you are right, the method should be self contained. 
Thanks for the suggestion. I will change the method and parameter name. 
Will send out a separate CR for additional Identifier read handler contract test

Reply:
calling method ```prune_properties_not_in_path``` because ```prune_properties``` already exists with the same parameters